connectors/sources/sharepoint_online.py

```diff
@@ -663,118 +719,259 @@ async def validate_config(self):
+            yield site_drive
+
+    async def drive_items(self, site_drive_id, max_data_age):
+        async for page in self.client.drive_items(site_drive_id):
+            for drive_item in page:
+                drive_item["_id"] = drive_item["id"]
+                drive_item["object_type"] = "drive_item"
+                drive_item["_timestamp"] = drive_item["lastModifiedDateTime"]
+
+                yield drive_item, self.download_function(drive_item, max_data_age)
+
+            self.update_drive_delta_link(drive_id=site_drive_id, link=page.delta_link())
+
+    async def site_list_items(
+        self, site_id, site_list_id, site_web_url, site_list_name
+    ):
+        async for list_item in self.client.site_list_items(site_id, site_list_id):
+            list_item["_id"] = list_item["id"]
+            list_item["object_type"] = "list_item"
+            list_item["_tempfile_suffix"] = os.path.splitext(
+                list_item.get("FileName", "")
+            )[-1]
+
+            content_type = list_item["contentType"]["name"]
+
+            if content_type in [
+                "Web Template Extensions",
+                "Client Side Component Manifests",
+            ]:  # TODO: make it more flexible. For now I ignore them cause they 404 all the time
+                continue
+
+            if "Attachments" in list_item["fields"]:
+                async for list_item_attachment in self.client.site_list_item_attachments(
+                    site_web_url, site_list_name, list_item["id"]
+                ):
+                    list_item_attachment["_id"] = list_item_attachment["odata.id"]
+                    list_item_attachment["object_type"] = "list_item_attachment"
+                    list_item_attachment["_timestamp"] = list_item[
+                        "lastModifiedDateTime"
+                    ]
+                    attachment_download_func = partial(
+                        self.get_attachment_content, list_item_attachment
+                    )
+                    yield list_item_attachment, attachment_download_func
+
+            yield list_item, None
+
+    async def site_lists(self, site_id):
+        async for site_list in self.client.site_lists(site_id):
+            site_list["_id"] = site_list["id"]
+            site_list["object_type"] = "site_list"
+
+            yield site_list
+
+    async def site_pages(self, url):
+        async for site_page in self.client.site_pages(url):
+            site_page["_id"] = site_page["GUID"]
+            site_page["object_type"] = "site_page"
+
+            for html_field in ["LayoutWebpartsContent", "CanvasContent1"]:
+                if html_field in site_page:
+                    site_page[html_field] = html_to_text(site_page[html_field])
+
+            yield site_page
+
+    def initiate_sync_cursor(self):
+        if not self._sync_cursor:
+            self._sync_cursor = {CURSOR_SITE_DRIVE_KEY: {}}
+
+        return self._sync_cursor
```

Comment:
Technically, it does make sense to do so because `nil`  can be considered as a successful operation. I assume I will change it anyway based on other feedback.